src/components/Banner/Banner.tsx

```diff
@@ -55,202 +59,210 @@ export interface BannerProps {
-      currentTarget.blur();
-      showFocus && this.setState({showFocus: false});
-    };
+    default:
+      return {
+        defaultIcon: newDesignLanguage
+          ? CircleInformationMajorFilled
+          : FlagMajorTwotone,
+        iconColor: newDesignLanguage ? 'base' : 'inkLighter',
+      };
+  }
+}
 
-    return (
-      <BannerContext.Provider value>
-        <WithinContentContext.Consumer>
-          {(withinContentContainer) => {
-            const {
-              icon,
-              action,
-              secondaryAction,
-              title,
-              children,
-              status,
-              onDismiss,
-              stopAnnouncements,
-            } = this.props;
-            let color: IconProps['color'];
-            let defaultIcon: IconProps['source'];
-            let ariaRoleType = 'status';
+function useBannerFocus(bannerRef: React.Ref<BannerHandles>) {
+  const wrapperRef = useRef<HTMLDivElement>(null);
+  const [shouldShowFocus, setShouldShowFocus] = useState(false);
 
-            switch (status) {
-              case 'success':
-                color = newDesignLanguage ? 'success' : 'greenDark';
-                defaultIcon = newDesignLanguage
-                  ? CircleTickMajorFilled
-                  : CircleTickMajorTwotone;
-                break;
-              case 'info':
-                color = newDesignLanguage ? 'highlight' : 'tealDark';
-                defaultIcon = newDesignLanguage
-                  ? CircleInformationMajorFilled
-                  : CircleInformationMajorTwotone;
-                break;
-              case 'warning':
-                color = newDesignLanguage ? 'warning' : 'yellowDark';
-                defaultIcon = newDesignLanguage
-                  ? CircleAlertMajorFilled
-                  : CircleAlertMajorTwotone;
-                ariaRoleType = 'alert';
-                break;
-              case 'critical':
-                color = newDesignLanguage ? 'critical' : 'redDark';
-                defaultIcon = newDesignLanguage
-                  ? CircleDisabledMajorFilled
-                  : CircleDisabledMajorTwotone;
-                ariaRoleType = 'alert';
-                break;
-              default:
-                color = newDesignLanguage ? 'base' : 'inkLighter';
-                defaultIcon = newDesignLanguage
-                  ? CircleInformationMajorFilled
-                  : FlagMajorTwotone;
-            }
-            const className = classNames(
-              styles.Banner,
-              status && styles[variationName('status', status)],
-              onDismiss && styles.hasDismiss,
-              showFocus && styles.keyFocused,
-              withinContentContainer
-                ? styles.withinContentContainer
-                : styles.withinPage,
-              newDesignLanguage && styles.newDesignLanguage,
-            );
+  useImperativeHandle(bannerRef, () => ({
+    focus: () => {
+      wrapperRef.current?.focus();
```

Comment:
Yay new features! (this is all good, I'm just excited that we're starting to dip our toes into these super-modern features)